bpf/trace_common.h

```diff
@@ -196,10 +204,16 @@ static __always_inline void get_or_create_trace_info(http_connection_metadata_t
 
     if (meta) {
         if (meta->type == EVENT_HTTP_CLIENT) {
+            tp_info_pid_t *in_tp = bpf_map_lookup_elem(&outgoing_trace_map, conn);
```

Comment:
Before this change the client code only looked for a server trace and if it doesn't find it it will generate the trace information later. Now we look if the TCP egress has setup trace info for us, if we find it we mark as having trace info, we must not regenerate it. If we have our own TCP egress info, but there's server incoming request that wrapped this client call, then we'll overwrite this info.

Reply:
I'd actually put that in a comment